### PR TITLE
Add blog post for Metalhead release v0.8

### PR DESCRIPTION
**Pull request by theabhirath**
Since Metalhead v0.8 is now tagged, I thought this would be a good explainer on a lot of the work that's happened in between v0.7 and v0.8.

cc @darsnack @ToucheSir for comments 😁

### CODE REVIEW

**Comment by theabhirath** [comment]
Bump on this if it's GTG?

File: blogposts/2023-06-07-metalhead-v0.8.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+Metalhead v0.8.0 is one of the largest releases of the library in recent times. Here's a list of all the improvements that have been accumulated in the library over the last few months:
+
+## More models
+
+Metalhead v0.8.0 ships with more exported models than any other previous Metalhead release. In particular, support for the following models was added:
+
+- [Inceptionv4, InceptionResNetv2 and Xception](https://github.com/FluxML/Metalhead.jl/pull/170)
+- [WideResNet, SE-ResNet and SE-ResNeXt](https://github.com/FluxML/Metalhead.jl/pull/174)
+- [Res2Net and Res2NeXt](https://github.com/FluxML/Metalhead.jl/pull/195)
+- [EfficientNetv2 and MNASNet](https://github.com/FluxML/Metalhead.jl/pull/198)
```
Comment (CarloLucibello) [inline blogposts/2023-06-07-metalhead-v0.8.md:16]: ViT is missing here

File: blogposts/2023-06-07-metalhead-v0.8.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+In Metalhead v0.7, support was added for pre-trained models for VGG and ResNets. v0.8.0 takes this further by adding support for Wide ResNets (an architecture previously not supported by Metalhead), certain configurations of ResNeXt and SqueezeNet. This makes it easier for users to get started with transfer learning tasks. We also now export the `backbone` and `classifier` functions, which return the feature extractor and classifier head portions of the model respectively. This should make it easier for users to hit the ground running.
```
Comment (CarloLucibello) [inline blogposts/2023-06-07-metalhead-v0.8.md:18]:
```suggestion
In Metalhead v0.7, support was added for pre-trained models for VGG and ResNets. v0.8.0 takes this further by adding support for Wide ResNets (an architecture previously not supported by Metalhead), certain configurations of ResNeXt, and SqueezeNet. This makes it easier for users to get started with transfer learning tasks. We also now export the `backbone` and `classifier` functions, which return the feature extractor and classifier head portions of the model respectively. This should make it easier for users to hit the ground running.
```

File: blogposts/2023-06-07-metalhead-v0.8.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+4. Squeeze-and-excitation layers were added, including an efficient squeeze-and-excite layer for future usage ([https://github.com/FluxML/Metalhead.jl/pull/174](https://github.com/FluxML/Metalhead.jl/pull/174)).
+5. `conv_bn` has now become `conv_norm`, with support for changing the normalisation layer ([https://github.com/FluxML/Metalhead.jl/pull/174](https://github.com/FluxML/Metalhead.jl/pull/174)). It also has changed default behaviour, with the bias for the convolution layer being set to `false` if the normalisation layers is enabled – this reflects the most common usage in most neural network models and so makes the user's job a little easier. There is also a `basic_conv_bn` layer that derives from the `conv_norm` layer which replicates the convolution-batch norm combination used in TensorFlow by default. In particular, the momentum value for BatchNorm is set to 3e-3.
+6. Inverted residual bottleneck blocks have been added, most notably `mbconv` and `fused_mbconv`, two commonly used variants in models like MobileNets and EfficientNets. ([https://github.com/FluxML/Metalhead.jl/pull/198](https://github.com/FluxML/Metalhead.jl/pull/198))
```
Comment (CarloLucibello) [inline blogposts/2023-06-07-metalhead-v0.8.md:31]: is this list related to Layers?

File: blogposts/2023-06-07-metalhead-v0.8.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+v0.8 takes this to the next level by separating out the model interfaces into three distinct interfaces. For the purpose of this article, we’ll call these the “high-level”, “mid-level” and “low-level” interfaces – and there have been improvements to all three of these.
+
+### Uniformity at the higher level interface
+
+The “high-level” interface caters to users who want a quick start and usually want to work with pre-trained models, either as feature extractors or to fine tune for transfer learning tasks. The notable improvement here has been that all of these functions now expose three keyword arguments mandatorily – `inchannels`, `nclasses`, and `pretrain`. The other big change is that there are no more default model configurations, doing away with ambiguous notation like `ResNet()`, which meant ResNet-50 in Metalhead v0.7. This work landed in [https://github.com/FluxML/Metalhead.jl/pull/190](https://github.com/FluxML/Metalhead.jl/pull/190).
+
+### New age training techniques at the mid level interface
```
Comment (CarloLucibello) [inline blogposts/2023-06-07-metalhead-v0.8.md:45]: age training?

**Comment by CarloLucibello** [comment]
would be nice to add a couple of code examples

File: blogposts/2023-06-07-metalhead-v0.8.md (hunk 3 above)
Comment (theabhirath) [reply blogposts/2023-06-07-metalhead-v0.8.md:31]: Yes, this is specific to `Layers`.

File: blogposts/2023-06-07-metalhead-v0.8.md (hunk 4 above)
Comment (theabhirath) [reply blogposts/2023-06-07-metalhead-v0.8.md:45]: Poorly worded - I'll change that.

**Comment by theabhirath** [comment]
> would be nice to add a couple of code examples

I avoided this deliberately. There have been issues in the past with old syntax lying around in blog posts not being updated causing errors for new users, which is a pain point for onboarding. Thus redirecting users to the docs is the best solution as that can be kept updated without much hassle in case the library changes something.

File: blogposts/2023-06-07-metalhead-v0.8.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+- [Inceptionv4, InceptionResNetv2 and Xception](https://github.com/FluxML/Metalhead.jl/pull/170)
+- [WideResNet, SE-ResNet and SE-ResNeXt](https://github.com/FluxML/Metalhead.jl/pull/174)
+- [Res2Net and Res2NeXt](https://github.com/FluxML/Metalhead.jl/pull/195)
+- [EfficientNetv2 and MNASNet](https://github.com/FluxML/Metalhead.jl/pull/198)
+- [The ViT model introduced in v0.7 is now more robust](https://github.com/FluxML/Metalhead.jl/pull/230) and comes with an option for [loading pre-trained weights on ImageNet](https://github.com/FluxML/Metalhead.jl/pull/235)
+
+In Metalhead v0.7, support was added for pre-trained models for VGG and ResNets. v0.8.0 takes this further by adding support for Wide ResNets (an architecture previously not supported by Metalhead), certain configurations of ResNeXt, and SqueezeNet. This makes it easier for users to get started with transfer learning tasks. We also now export the `backbone` and `classifier` functions, which return the feature extractor and classifier head portions of the model respectively. This should make it easier for users to hit the ground running.
```
Comment (ToucheSir) [inline blogposts/2023-06-07-metalhead-v0.8.md:19]:
```suggestion
In Metalhead v0.7, support was added for pre-trained models for VGG and ResNets. v0.8.0 takes this further by adding support for Wide ResNets (an architecture previously not supported by Metalhead), certain configurations of ResNeXt, and SqueezeNet. This makes it easier for users to get started with transfer learning tasks. We also now export the [`backbone` and `classifier`](https://fluxml.ai/Metalhead.jl/v0.8/api/utilities) functions, which return the feature extractor and classifier head portions of the model respectively. This should make it easier for users to hit the ground running.
```

File: blogposts/2023-06-07-metalhead-v0.8.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+Metalhead is always looking for contributors to help with adding pre-trained weights for the models. To know how you can help with this effort, please check out the contributor’s guide in the documentation. We will be happy to help you work through any issues you may encounter!
+
+## A `Layers` module to make it easier to build custom models
+
+Previously, Metalhead v0.7 introduced the `Layers` module but it was not publicly documented as the internals were still being polished. With Metalhead v0.8.0, it has reached a point of stability. The `Layers` module exposes functionality that allows users to build custom neural network models very easily. Some notable improvements are:
```
Comment (ToucheSir) [inline blogposts/2023-06-07-metalhead-v0.8.md:25]:
```suggestion
Previously, Metalhead v0.7 introduced the `Layers` module but it was not publicly documented as the internals were still being polished. With Metalhead v0.8.0, it has reached a point of stability. The [`Layers`](https://fluxml.ai/Metalhead.jl/v0.8/api/layers) module exposes functionality that allows users to build custom neural network models very easily. Some notable improvements are:
```

File: blogposts/2023-06-07-metalhead-v0.8.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+
+The “low-level” interface allows engineers and researchers to iterate on model design quickly without thinking too much about the building blocks. In particular, multiple dispatch means that functions that need to be re-written again and again in Python libraries can simply be unified under one function in Julia.
+
+One of the major changes Metalhead v0.8 introduces is the concept of *builders.* These are functions that return closures over the **stage and block indices**. These two numbers together index into a block completely (the stage index represents which stage of the large model is being referred to, and the block index finds the specific block inside the stage). Builders have allowed writing functions that construct blocks purely based on the stage and block indices, abstracting away construction of the layers as well as details like calculations for strides, output feature maps or probabilities for stochastic depth/DropBlock. These builders have been used to rewrite the underlying implementations of the three largest CNN families supported by Metalhead and that are used commonly – ResNets, MobileNets and EfficientNets.
```
Comment (ToucheSir) [inline blogposts/2023-06-07-metalhead-v0.8.md:62]:
```suggestion
One of the major changes Metalhead v0.8 introduces is the concept of *builders.* These are functions that return closures over the **stage and block indices**. These two numbers together index into a block completely (the stage index represents which stage of the large model is being referred to, and the block index finds the specific block inside the stage). Builders have allowed writing functions that construct blocks purely based on the stage and block indices, abstracting away construction of the layers as well as details like calculations for strides, output feature maps or probabilities for stochastic depth/DropBlock. These builders have been used to rewrite the underlying implementations of the three largest CNN families supported by Metalhead and that are used commonly – [ResNets](https://fluxml.ai/Metalhead.jl/v0.8/api/resnet/#Block-builders), MobileNets and EfficientNets.
```

File: blogposts/2023-06-07-metalhead-v0.8.md [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+
+The biggest beneficiary of this has been the MobileNet and EfficientNet model families, which are constructed from a single function using nothing but a uniform configuration dictionary format. Yep, you read that right. That’s six different models expressed using a single underlying function which is just about sixty lines of code. As a user, all you need to do is change the configuration dictionary and watch the magic happen. This means that conceptualising new models has become as simple as visualising the model structure as a dictionary of configurations (in short, just the novelties), and then watching the function take your configurations and produce the model.
```
Comment (ToucheSir) [inline blogposts/2023-06-07-metalhead-v0.8.md:64]:
```suggestion
The biggest beneficiary of this has been the MobileNet and EfficientNet model families, which are constructed from a [single function](https://github.com/FluxML/Metalhead.jl/pull/200) using nothing but a uniform configuration dictionary format. Yep, you read that right. That’s six different models expressed using a single underlying function which is just about sixty lines of code. As a user, all you need to do is change the configuration dictionary and watch the magic happen. This means that conceptualising new models has become as simple as visualising the model structure as a dictionary of configurations (in short, just the novelties), and then watching the function take your configurations and produce the model.
```
Or reference the PR directly like you did for the other work done.

File: blogposts/2023-06-07-metalhead-v0.8.md [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+
+There were improvements for the ResNet family of models as well: the stride and feature map calculations through the ResNet block builders are now callbacks, and can thus be easily modified by users if they wish to define custom stride/feature map calculations. To know more about builders and how they work for the different model families, check out the documentation! We are still working on adding documentation for all of the features, so if you find something that's unexplained, open an issue and we will be happy to assist you as well as to improve the documentation!
```
Comment (ToucheSir) [inline blogposts/2023-06-07-metalhead-v0.8.md:66]: I'm guessing we don't yet have docs to link to for this?

**Review by ToucheSir — APPROVED** [review]
Mostly LGTM. Some suggestions for links you might want to add.

File: blogposts/2023-06-07-metalhead-v0.8.md (hunk 7 above)
Comment (theabhirath) [reply blogposts/2023-06-07-metalhead-v0.8.md:62]: That page has a misleading title 😅 Here we are taking about the model builders, not the block builders.

File: blogposts/2023-06-07-metalhead-v0.8.md (hunk 9 above)
Comment (theabhirath) [reply blogposts/2023-06-07-metalhead-v0.8.md:66]: Yeah, we don't. I hope we can remedy that soon 😅, but for now I'd rather have people know about v0.8 and start using the features we _have_ documented.

**Comment by theabhirath** [comment]
Bump on merge if it's okay?

**Comment by theabhirath** [comment]
Is it okay for me to link this on my social media now? I do want to let people know that there's been significant improvements they should be using 😅

**Comment by ToucheSir** [comment]
Absolutely, go for it!
